src/test/java/edu/ucla/library/iiif/fester/fit/PostCsvFIT.java

```diff
@@ -584,7 +600,130 @@ public final void testVideoIiifPresApiV2(final TestContext aContext) {
+
+                    LOGGER.error(exception, exception.getMessage());
+                    aContext.fail(exception);
+                }
+            });
+        }
+
+       /**
+         * Tests submitting a CSV with an audio object.
+         *
+         * @param aContext A test context
+         */
+        @Test
+        public final void testAudio(final TestContext aContext) {
+            final Async asyncTask = aContext.async();
+
+            postCSV(AUDIO_CSV, Constants.IIIF_API_V3, post -> {
+                if (post.succeeded()) {
+                    final HttpResponse<Buffer> response = post.result();
+                    final int statusCode = response.statusCode();
+                    final String statusMessage = response.statusMessage();
+
+                    if (statusCode == HTTP.CREATED) {
+                        final Optional<JsonObject> optJsonObject = checkS3(AV_ARK, false);
+
+                        if (optJsonObject.isPresent()) {
+                            try {
+                                final JsonObject expected = readJsonFile(AUDIO_MANIFEST);
+                                final JsonObject found = optJsonObject.get();
+
+                                aContext.assertTrue(TestUtils.manifestsAreEffectivelyEqual(expected, found));
+                            } catch (final IOException details) {
+                                LOGGER.error(details, details.getMessage());
+                                aContext.fail(details);
+                            }
+                        } else {
+                            aContext.fail(LOGGER.getMessage(MessageCodes.MFS_154, AUDIO_CSV));
+                        }
+                        complete(asyncTask);
+                    } else {
+                        aContext.fail(LOGGER.getMessage(MessageCodes.MFS_039, statusCode, statusMessage));
+                    }
+                } else {
+                    final Throwable exception = post.cause();
+
+                    LOGGER.error(exception, exception.getMessage());
+                    aContext.fail(exception);
+                }
+            });
+        }
+
+        /**
+         * Tests submitting a CSV with an audio object that is missing metadata.
+         *
+         * @param aContext A test context
+         */
+        public final void testAudioMissingMetadata(final TestContext aContext) {
+            final Async asyncTask = aContext.async();
+
+            postCSV(AUDIO_CSV_MISSING_METADATA, Constants.IIIF_API_V3, post -> {
+                if (post.succeeded()) {
+                    final HttpResponse<Buffer> response = post.result();
+                    final int statusCode = response.statusCode();
+                    final String statusMessage = response.statusMessage();
+
+                    aContext.assertEquals(statusCode, HTTP.BAD_REQUEST);
+                    aContext.assertTrue(statusMessage.contains(AV_MISSING_ERROR));
```

Comment:
First of all, I believe I wrote this incorrectly in the video tests, because `"missing either width, height, or duration"` is not a substring of `"Found item '{}' with format '{}' in CSV '{}', but is missing either width, height, duration, or A/V access URL"`. So I don't understand why this assertion isn't failing.

Second, I think the audio tests should use a different message, since audio won't have width or height.

Reply:
Split audio and video messages, tweaked wording of video message